### ISSUE TITLE
Misleading error message when extending type parameter

### ISSUE THREAD

**Issue by rotu** [microsoft/TypeScript#57037]
### 🔎 Search Terms

"statically known members."
"could be instantiated with a different subtype of constraint 'object'."

### 🕗 Version & Regression Information

- This changed between versions 3.6.2 and 3.7.5

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=3.7.5#code/JYOwLgpgTgZghgYwgAgKIA84FsAOAbCAHgBVkJ1IQATAZ2QHsAjAKwgTAD4yKJq7SA3gF8AUEA

### 💻 Code

```ts
interface Example<T extends object> extends T {}
```

### 🙁 Actual behavior

I get the following message:

> Interface 'Example<T>' incorrectly extends interface 'T'.
>   'Example<T>' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'object'.

This error message is misleading. The interface does *not* vary with `T` and since it declares no new members, is completely compatible with `T`.

### 🙂 Expected behavior

I expect a descriptive message. Prior to 3.6.2, the error message was more helpful:
> An interface can only extend an object type or intersection of object types with statically known members.

This is still the error you get if the type variable `T` does not have the `extends` constraint.

**Comment by rotu** [comment]
Relatedly, you can't extend other type expressions that aren't type identifiers or instantiations of generics, and the error message is similarly dense: #31843
